internal/policy/list.go

```diff
@@ -0,0 +1,104 @@
+// Copyright The gittuf Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package policy
+
+import (
+	"context"
+
+	"github.com/gittuf/gittuf/internal/gitinterface"
+	"github.com/gittuf/gittuf/internal/tuf"
+)
+
+type DelegationWithDepth struct {
+	Delegation tuf.Rule
+	Depth      int
+}
+
+// ListRules returns a list of all the rules as an array of the delegations in a
+// pre-order traversal of the delegation tree, with the depth of each
+// delegation.
+func ListRules(ctx context.Context, repo *gitinterface.Repository, targetRef string) ([]*DelegationWithDepth, error) {
```

Comment:
while this may be inscope because I reorganized the func def, maybe it belongs elsewhere. FWIW, I think we need to rework our list / show commands because these shouldn't really be separate commands.